src/ftest.jl

```diff
@@ -0,0 +1,92 @@
+"""A helper function to determine if mod1 is nested in mod2"""
+function issubmodel(mod1::LinPredModel, mod2::LinPredModel)
+    mod1.rr.y != mod2.rr.y && return false # Response variables must be equal
+
+    # Now, test that all predictor variables are equal
+    pred1 = mod1.pp.X
+    npreds1 = size(pred1, 2)
+    pred2 = mod2.pp.X
+    npreds2 = size(pred1, 2)
+    # If model 1 has more predictors, it can't possibly be a submodel
+    npreds1 > npreds2 && return false 
+    
+    @inbounds for i in 1:npreds1
+        var_in_mod2 = false
+        for j in 1:npreds2
+            if view(pred1, :, i) == view(pred2, :, j)
+                var_in_mod2 = true
+                break
+            end
+        end
+        if !var_in_mod2
+            # We have found a predictor variable in model 1 that is not in model 2
+            return false 
+        end
+    end
+
+    return true
+end
+
+"""
+    ftest(mod::LinPredModel...)
+
+For each sequential pair of linear predictors in `mod`, perform an F-test to determine if the first one fits significantly better than the second
```

Comment:
Please break lines at 92 chars, even inside docstrings. Missing ending period. Also the function does more than testing, it returns a series of statistics which should be mentioned.